client/src/components/Navbar/Navbar.js

```diff
@@ -1,21 +1,105 @@
-import React from 'react';
+import React, { Component } from "react";
 import "./Navbar.css";
-import Logo from "../Logo/Logo"
+import Logo from "../Logo/Logo";
+import Burger from "../Burger/Burger";
+import ProfilePicture from "../ProfilePicture/ProfilePicture";
+import NavLink from "../NavLink/NavLink";
+import Icon from "../Icon/Icon";
+import PropTypes from "prop-types";
 
-// Function based React Component
-const Navbar = (props) => {
+// Class Based React Component
+class Navbar extends Component {
+  constructor(props) {
+    super(props);
+    // Default CSS class to apply to the Component
+    this.state = {
+      condensedMenuActive: false,
+      condensedMenu: "",
+      // pageMark controls the triangle that marks which page the user is on.
+      // Passing 0-3 as a prop into this component controls where it displays.
+      pageMark: ["", "", "", ""],
+      // menuItems are careated in order of the following array.
+      menuItems: ["My Directory", "Community", "Forum", "Book a Trip"],
+      // The link end points for each menu item.
+      menuLinks: ["/", "/", "/", "/"],
+      // Each item in this array is a set for the dropdown on each menu link.
+      // This is an example for one drop menu item: { text: 'Explore People', to: '/' }
+      menuDropDown: [[], [], [], []]
+    };
+  }
 
-  // Default Class to apply to Component
-  let classList = `Navbar`;
+  // Handles activating the hamburger animation and displays the menu.
+  handleCondensedMenuClick = () => {
+    this.setState({ condensedMenuActive: !this.state.condensedMenuActive })
+    if (this.state.condensedMenuActive === false) {
+      this.setState({ condensedMenu: "Navbar__show" });
+      this.setState({ pageMark: ["", "", "", ""] });
+    } else {
+      this.setState({ condensedMenu: "" });
+      this.setPageMarker();
+    }
+  }
```

Comment:
Can you simplify this?

Reply:
One way would be to use the `.Navbar__show` class - if `condensedMenuActive` is `true`, then add it in the `className` attribute below using a ternary, otherwise don't add it. In your CSS, if the class is there, then the menu is visible, otherwise it isn't.

Reply:
Something to be careful of though is calling `setState` twice back-to-back like that. `setState` is asynchronous and this will trigger two renders. Can you merge the two `setState` calls?

Reply:
Indeed! I already caught this one, actually, when I moved the function down to the render area this afternoon. I kept working on things after I sent it back to you.

Reply:
This whole if/else statement is gone now.